### ISSUE TITLE
GHA Data Visualisations 2021

### ISSUE THREAD

**Issue by edwinmp** [devinit/DIwebsite-redesign#1028]
**Resources**

- [Specification Document](https://docs.google.com/document/d/1NzypaXqLqvyBAN4zQwL8PM9X1x8n-21HeHV767CgVDQ/edit)

**Comment by edwinmp** [comment]
Hi @alice-mcandrew-development-initatives I've created a [prototype](https://devinit.org/admin/pages/2450/edit/preview/) on which all the other charts are going to be based ... they'll most likely be bar charts (based on the excel examples), but I'm more interested in feedback on the multi-select comparison feature ... I had to draw from widgets used on the spotlights for that...

![image](https://user-images.githubusercontent.com/5672438/126482221-0d3d77d0-9bb2-43f7-b28b-3817c7c54595.png)

**Comment by alice-mcandrew-development-initatives** [comment]
Thanks, @edwinmp! It's really useful to have a steer on what you're looking for feedback on in particular, as well having the spec upfront in the ticket to refer back to. I'll take a look and come back with comments later today.

So you're aware, I got a preview error when I tried the link provided for the prototype:
![image](https://user-images.githubusercontent.com/55877063/126612319-2bedac1a-64c8-4e33-b04f-69914f898612.png)

I'm able to view the prototype by removing the 'preview' at the end of the URL, so using: https://devinit.org/admin/pages/2450/edit/

But those without cms admin rights wouldn't be able to do this, so that'll be something to check when sharing with the engagement and analyst team.

**Comment by alice-mcandrew-development-initatives** [comment]
Just to confirm, you’re working out how the bar charts will work in general (as that's relevant to all three) then once that’s agreed you'll proceed to flesh out the three charts as outlined in the spec?

I’ve focused my feedback on the desktop version initially. Do let me know what is and isn’t helpful in the below (including letting me know anything that’s unclear) so that I make sure I’m getting you what you need to help you to refine this.

**Comments on multi-select comparison for desktop:**

1. Suggest the menu title is ‘Select donors’ as plural (to indicate they can choose to show more than one if they’re using the multi-select comparison feature)
2. Suggest the menu title is emboldened (as it’s the call-to-action we want people to respond to)
3. Suggest the menu default is ‘Ten largest donors’ (as we’re not showing all donor countries)
4. Suggest pebbles are the same font size as menu (to indicate they are related)
5. Really nice touch with the animation – that feedback adds a great dynamic feel to the interaction!
6. Sometimes when I added a country they appeared at the bottom and sometimes they appeared at the top and I couldn’t work out if there was a reason this was happening – suggest that new countries selected always appear at the top
7. Suggest that the order of the pebbles always matches the order of the bars
8. Sometimes when I clicked on the ‘x’ on the pebble the bar disappeared but the pebble retains in place
9. I wonder if there is a way we can make it obvious which among the ten donors are left to add via the menu (without looking back/forth between menu and pebbles) when users have taken the route of adding donors via the multi-select comparison

**Comments on other aspects for desktop:**

1. Suggest the default view is arranged so that the largest donor appears at the top
2. Suggest axes title/labels are at the top, rather than bottom, so that if on a smaller screen you have it first, without needing to scroll down to see it
3. Include ‘US$ millions’ or ‘US$ billions’ as an axis title
4. Suggest you remove the grey background beyond the bars (so it doesn’t appear like its X out of Y)
5. The dotted line that appears when you hover over the bars is very thoughtful and provides valuable feedback – I wonder if instead of a dotted line we could shade the bar, or stack element when this evolves, darker on hover (to make it jump out visually); if proceeding with the dotted line approach then I’d suggest that we end the line at the end of the bar or stack element, rather than the edge of the chart grid
6. Suggest we consider removing the coloured dot (as they will all be the same colour presumably or responsive to which element of the bar you’re hovering over) and the word ‘Countries’ (which is not strictly necessary) in the tooltip for the chart
7. Suggest that we have the arrow rather than hand when hovering over bars (as we don’t want to indicate the user can take action if they select this, as that isn’t possible)
8. Suggest we use ‘Save as PNG’ for the tooltip for the image
9. Is it possible to label the PNG title as the name of the chart?

**Questions for the team:**

1. Who is providing the heading and chart title etc? Is this to be drafted alongside other supporting copy?
2. Presumably we want pebbles to appear in the order selected, rather than having it so that when you add a new pebble the pebbles and bars re-order so we’re always showing the largest donor first?
3. Are country labels set? Why does Germany include ‘(FRG)’? Can this difference be commented on in notes/source? Also, can we use ‘US’ and ‘UK’ instead of writing those out so pebbles are smaller?

**Comment by edwinmp** [comment]
@alice-mcandrew-development-initatives here's the updated chart for the Funding Channels data viz ... had this ready before I saw this feedback. I'll implement your feedback while you take a look at the updated version ...
https://devinit.org/resources/gha-data-visualisations-2021/

**Comment by kirstylazer** [comment]
@edwinmp thank you so much for all your work on these we all agree they are looking really good! I have collated the teams comments and am sharing them below - please let us know if you have any questions or need any clarification.

General Comments
1.	Embolden menu title (it is call to action we want people to respond to)
2.	Change pebbles to same font size as the menu (to indicate they are related)
3.	Dotted line that appears when hovering is a good touch, ensure that does not extend above the top of the bar (it currently does on the recipients chart, if we could change this). Also darken colour of element in hover state (or keep colour and reduce opacity of other sections that are not in hover state)
4.	Change to arrow as opposed to hand when hovering over bars (as don’t want to indicate user can take action if they select this as it isn’t possible
5.	Increase the font size of the legend and move to the top of the chart, this will put it nearer to other chart elements that users can interact with, making it more obvious that can select/de-select parts of the legend
6.	For tooltip to save image change to ‘Save as PNG’ as opposed to ‘saveAsImage’ which is what we have currently
7.	Label PNG title as the name of the chart
8.	Add totals to the top of each bar for any charts showing volume (i.e. the top of each bar for donor’s chart when showing volumes and the top of each bar for the recipient’s chart).
9.	Change colour of charts – have same colour scheme for donor’s and channels chart to indicate linkage, and a different colour scheme for recipient’s chart
10.	Change so that for proportions charts when de-select parts of the legend that the y-axis rescales (so can detect smaller % differences) – as is the case for volumes chart currently. Also ensure for donor’s volumes chart (to be added) that y-axis rescales.

Donors Chart
1.	Change the default chart to one showing volumes of funding to each category, and add a toggle to switch to proportions chart (i.e. the current version). Team agree that volumes is probably more relevant for users.
2.	Change menu title to ‘Select donors’ to indicate can select more than one donor
3.	Change menu default to ’20 largest donors’ as not actually showing all donor countries
4.	If a user tries to select an item from the menu but they already have two donors selected can we provide an error message in red font to the right of the menu: ‘You can compare two donors. Please remove one before adding another.’
5.	When two donors are selected, can the dotted line appear on a bar rather than between the bars to avoid confusion as to which is in hover state
6.	Change so tooltip is organised as follows: ‘[Country], [Year] / [Stack element] / X% (US$X)’
7.	‘HA’ should be written out in full as ‘Humanitarian assistance’ as it’s not a common acronym – although keep HA in legend label otherwise too wordy (clarify in chart title what HA refers to).

Channels Chart
1.	Change to RCRC in legend label, and explain in notes, otherwise legend too wordy

Recipients Chart
1.	When change chart breakdown option from one to the other can we change it so chart does not revert to showing all top 10 recipients? So if one country selected can change between breakdown options easily
2.	Add axis label – US$ millions
3.	Use consistent decimal places in tooltip – change all to 2 decimal places, currently showing too many decimal places
4.	Order toggle dropdown from largest to smallest recipient (i.e. Syria at top)
5.	Change first menu title to ‘Select recipient’
6.	Change second menu title to ‘Choose breakdown’
7.	Change so that we just show information relevant to the element of the stack that is being hovered over (for consistency with functioning of donor chart)
8.	Change to ‘By type of organisation receiving funding’, rather than ‘By Recipient Organisation Type’, for clarity
9.	Use ‘DRC’ (as a common acronym) and ‘Syria’ in the menu - we can explain in notes below
10.	Use ’10 largest recipients’ as default menu option as not actually all recipient countries
11.	Use ‘US’, ‘UK’ and ‘UAE’ as legend labels - again, we can explain in notes below
12.	Use ‘NGOs and CSOs’ and ‘RCRC’ in legend labels
13.	Use ‘UN multilateral organisations’ in legend labels so sentence case

**Comment by edwinmp** [comment]
@davidebukali if there's nothing else pending on this, please close it.

**Comment by alice-mcandrew-development-initatives** [comment]
@davidebukali I'm re-opening this as I've noticed some changes that need to be made before we publish.

- [ ] Please can you amend so that for each chart (from x axis upwards) we are using the following order:

1. #eb642b
2. #f6bb9d
3. #f18e5e
4. #d85b31
5. #973915
6. #f4a57c

(The first five are ordered as outlined in the DI brand book but I am suggesting we use Orange 9 for the sixth colour as Orange 6 is very pale so may not be easily discernable) 

- [ ] Please can we rotate the y-axis title -90 degrees, so it looks more like this:

![image](https://user-images.githubusercontent.com/55877063/135297135-ed4d5074-3e17-443f-ad88-a9d54dbc6625.png)

- [ ] Please change the tooltip format for 'Compare volumes of international humanitarian assistance from, and how it is allocated by, the largest donors' to:

[Donor], [Year]
[Legend category]: [**US$X million**] OR [**X.XX%**] 

(We currently have a dash and extra spacing that needs to be closed up, as well as inconsistency depending on whether you are looking at one or all donors)

- [ ] Please change the tooltip format for 'Compare how the largest donors channel international humanitarian assistance' to:

[Donor], [Year]
[Legend category]: [**X.XX%** (US$X million)]

(We currently have a dash, extra spacing that needs to be closed up, an 's' on the end that needs deleting and bold on the U$ figure which is not needed, as well as inconsistency depending on whether you are looking at one or all donors)

- [ ] Please change the tooltip format for 'Explore funding to the largest country recipients of international humanitarian assistance, their leading donors and types of recipient organisations' to:

a) If viewing by donor...

[Legend category], [Year]
[Recipient selected in dropdown]: [**(US$X million**]

b) If viewing by type of organisation receiving funding...

[Year]
[Legend category]: [**US$X million**]

(This will be more consistent with the other two charts, factoring in the different type of info shown)

Let's have a call on Slack if anything I've outlined above is not clear.

**Comment by alice-mcandrew-development-initatives** [comment]
Sorry to add to the list - one more request from Nik R and Dan W:

- [ ] Use same colours for types of organisations in 'Compare how the largest donors channel international humanitarian assistance' (those that appear in the legend) the same as for 'Explore funding to the largest country recipients of international humanitarian assistance, their leading donors and types of recipient organisations' (those that appear in the legend when you select the 'By type of organisation receiving funding')

**Comment by k8hughes** [comment]
Ready for review @alice-mcandrew-development-initatives  or Louise....
